jscomp/syntax/ppx_entry.ml

```diff
@@ -270,9 +288,12 @@ let rec unsafe_mapper : Ast_mapper.mapper =
         | Pexp_extension (
             {txt = ("bs.re" | "re"); loc} , payload)
           ->
+          if validate_regexp payload then
           Exp.constraint_ ~loc
             (Ast_util.handle_raw loc payload)
-            (Ast_comb.to_js_re_type loc)            
+            (Ast_comb.to_js_re_type loc)
```

Comment:
I think `handle_raw` already has the logic to destruct it as a string, can we push the validation logic there, maybe you can pass the validation function as an argument